napari/_qt/qt_viewer.py

```diff
@@ -788,7 +789,17 @@ def on_resize(self, event):
         self.viewer._canvas_size = tuple(self.canvas.size[::-1])
 
     def _process_mouse_event(self, mouse_callbacks, event):
-        """Called whenever mouse pressed in canvas.
+        """Add properties to the mouse event before passing the event to the
+        napari events system. Called whenever mouse pressed in canvas.
```

Comment:
this docstring is a little outdated.  this method is actually called whenever the mouse does _anything_ ... even just move slightly.  So this method is getting called approximately all the time.  I can't find the issue at the moment, but I recall that performance in this specific function is already causing really high CPU usage.  Nothing here looks to terribly costly, but just wanted to flag that.  If this stuff is only necessary on mouse click, then we might want to gate those calculations on a specific event type?

Reply:
Hmm yeah, good point .... folks might want the `ray` available to them during move though, for example hovering over a point to see its value/ label in 3D

Reply:
i just did some quick tests, and for me, this code added about 18% additional overhead... (went from about 1.1ms to 1.3 ms).  The absolute times are still pretty tiny, so probably fine to just move forward with it.  But something to think about.

Reply:
Sounds tolerable for now, but agreed something to continue to monitor closely

Reply:
We could/should throttle this call to 60/30fps? That might help bring down the CPU usage?

Reply:
Obviously, could happen in a new PR.

Reply:
yep, and yep

Reply:
> this docstring is a little outdated. this method is actually called whenever the mouse does anything ... even just move slightly. So this method is getting called approximately all the time. I can't find the issue at the moment, but I recall that performance in this specific function is already causing really high CPU usage. Nothing here looks to terribly costly, but just wanted to flag that. If this stuff is only necessary on mouse click, then we might want to gate those calculations on a specific event type?

Thanks for flagging this, @tlambert03 ! I have updated the docstring. I agree that it could make sense to set an upper limit on the frequency this is called.